### ISSUE TITLE
unable to songs or artists by name - playlists work

### ISSUE THREAD

**Issue by andocromn** [stevenleeg/geemusic#155]
I recently setup geemusic.  it seems to be working because I can ask Alexa to start a playlist and that works fine.  However if I ask for a song or artist that is in my library it responds with "Sorry, I couldn't find that song" (or artist depending on what you ask).  The debug output shows "Fetching song the pot by tool" so the data is getting to but its not finding the music.  I've tested with the Service Simulator in the Alexa Dev center too and the results are the same.

Any help is appreciated, let me know if more info is needed

**Comment by TheModdersDen** [comment]
Same here... I'm working on an attempt at a fix for this, based off of the code of PR #114. It's nearing perfection, though. I'll make a PR, and post here to let you know if and or when I finish it. :)

**Comment by bspesq** [comment]
I'm in the same boat.  Please let me know if you need help or testers for the code.

**Comment by famewolf** [comment]
Ditto.  Subscribed to this issue.

**Comment by andocromn** [comment]
Its worth noting that I have music in my google music library that I do not believe is available with Google's unlimited music service.  For example the last time I was at my friend's house with a google home (and the music subscription) It was not able to find the band Tool.  We found that none of their works were available through the subscription.

It would be good if someone could test whether or not Geemusic is able to play uploaded music which is not available through the subscription

**Comment by TheModdersDen** [comment]
@andocromn That is what my comment mentions. I have a Google Play Music library of over 10,000 uploaded songs that I have locally on my computer as well. 

I'm currently working on uploading my code for all to tinker, experiment with, and test. Note, I'm fairly new to Python, but I did get songs from my library playing, in addition to playlists. 

Despite the above, certain features like radios seem to be broken, both in my fork of this project, and the original code (this is coming from a free account, though. So that may be the problem).

Please remember that my code is heavily based on @andrewdavisdev's code. I give credit where credit is due, and though I modified the code from andrewdavisdev to work with the most "up-to-date" branch, it is still mostly his code.

I also have plans to implement a way to have the GeeMusic service re-scan one's library. This is currently a WIP. I'm open to any suggestions/solutions for getting this to work.

**Comment by famewolf** [comment]
When in my automobile playing google music via android auto I'm pretty much able to play whatever I want...perhaps a similar method could be used for free accounts?

**Comment by bspesq** [comment]
I did not see your fork before.  I have tested your code and get most functionality now.  Thanks!
I think the only thing not working is two or three part queries, meaning "play {something} by {someone}".  Debug output makes me think it's not parsing the "by" and just glomming it all together to search for a title?

```
10:39:24 web.1  | --------------------------------------------------------------------------------
10:39:24 web.1  | DEBUG in selection [/home/gmusic/geemusic/intents/selection.py:136]:
10:39:24 web.1  | Fetching song toto by Africa
10:39:24 web.1  | --------------------------------------------------------------------------------
10:39:24 web.1  | --------------------------------------------------------------------------------
10:39:24 web.1  | DEBUG in music [/home/gmusic/geemusic/utils/music.py:158]:
10:39:24 web.1  | get_song() : name: Africa toto
10:39:24 web.1  | --------------------------------------------------------------------------------
```

**Comment by TheModdersDen** [comment]
Yeah, I know. As I mentioned, I'm new to Python, but not to programming. If anyone has an idea for a fix (and preferably some code), let me know. 

My code is here: https://github.com/TheModdersDen/geemusic

NOTE: Please read the instructions again, especially the ones about the environment variables. You need to set a new variable in order to get the program to run. If you don't do this, the program will crash.

The new variable is "USE_LIBRARY_FIRST" (w/o quotes). Set it to true if you want to use your library. Set it to "False"

**Comment by famewolf** [comment]
Implemented your commits prior to your post and found out the hard way about the environment variable but figured it out from the logs.  Seems to work well for the basics although "play some music" still appears broken where theoretically it could still be reworked to pull randomly from library just like "play some music by xxxx" works.  Great job!

**Comment by TheModdersDen** [comment]
Glad to hear you got it working! My winter break starts the 18th of this month, so I should be able to work on this in my spare time... :)

**Comment by digiltd** [comment]
Brilliant, thanks, not sure what you did, I battled for hours with @andrewdavisdev's fork and couldn't get it to work. Well done. Possible to do a PR to this repo?

**Comment by TheModdersDen** [comment]
I'm holding off doing a PR until I can fix some of the issues mentioned above with my fork. At the time of writing the issues include:

- Can't play a specific song by an artist (it just jumps to the first song with the matched title). In other words, "play {something} by {someone}" fails to work.
- Radios don't work (I think I may have a fix for this, though... Got to do a bit more looking into it, however)

I also have been thinking of features for later stages of development, also mentioned above. I only have one in mind so far: a feature to re-scan the user's music library (as one needs to restart the server for this to happen).

**Comment by famewolf** [comment]
In the pie in the sky list...with alexa I have an ifttt hook that every time I play a song it adds an entry to a google drive database so I can later go review all the songs for ones I might want to download...if I had a method to have the info saved to something I can get my hands on that would be really helpful.  I'm using  a heroku instance if that makes a difference.

**Comment by TheModdersDen** [comment]
I'll keep that in mind. Does the IFTTT hook not work with GeeMusic?

**Comment by famewolf** [comment]
ifttt only has hooks for songs played on "alexa/amazon music", spotify and youtube..nothing for any skill or even pandora.

**Comment by TheModdersDen** [comment]
OK. I'll see what I can do in regards to that.

**Comment by TheModdersDen** [comment]
Alright. So, after a lot of fiddling, I was able to get the GeeMusic program to play a song by an artist. However, I could not get free accounts to support radios. I noticed that the gmusicapi library has been recently updated to fix this, but I don't know how to implement that here.

Due to errors when one tries to use radios on a free account, please avoid trying to use radios with free accounts. Your program **should** still work even if you try to use radios, but I can't guarantee anything...

I will also be discontinuing my development of this project, after reading that Amazon has discontinued their music storage service (which is one of the reasons as to why I bought an Echo Dot). I also don't have the ability to host this service on a dedicated platform, so that also factors into my decision (in addition to my novice skills at Python hindering my abilities to develop).

Despite all of this, I have updated my code on my repo to allow for free user accounts to play music (and play songs by artists). I will be making a pull request shortly after reviewing some recent code changes in this main repository...

**Comment by laurilarjo** [comment]
@TheModdersDen @famewolf I tried the code also and it doesn't work. And looking at the code I don't see how it ever could have worked for example for "play artist cher", because it always seems to require exact match being said "artisti - song name". So is the approach taken in this branch even viable? Or is it going to be too hard to make it flexible enough for your own songs?

**Comment by TheModdersDen** [comment]
@laurilarjo Sorry to say this, but I've discontinued my work on this project. I no longer have an Alexa device and therefore cannot do any more testing. Also, I'm not as skilled in Python as many of the other developers here...
